qcodes/config/config.py

```diff
@@ -121,26 +123,35 @@ def update_config(self):
         """
         config = copy.deepcopy(self.defaults)
         self.current_schema = copy.deepcopy(self.defaults_schema)
-
-        if os.path.isfile(self.home_file_name):
-            home_config = self.load_config(self.home_file_name)
-            config = update(config, home_config)
-            self.validate(config, self.current_schema,
-                          self.schema_home_file_name)
-
-        if os.path.isfile(self.env_file_name):
-            env_config = self.load_config(self.env_file_name)
-            config = update(config, env_config)
-            self.validate(config, self.current_schema,
-                          self.schema_env_file_name)
-
-        if os.path.isfile(self.cwd_file_name):
-            cwd_config = self.load_config(self.cwd_file_name)
-            config = update(config, cwd_config)
+        self.config_file_path = path or self.config_file_path
+        if self.config_file_path is not None:
+            config_file = "{}/{}".format(self.config_file_path, self.config_file_name)
```

Comment:
please use `os.path.join` instead of `/` and `format`. let the OS define the file separator.

Reply:
yup, that would be better, I just copied the original logic from likes 72 and 81 but will update them all now.